modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -7399,6 +7417,34 @@ public void testGetScopeScopeKeyObjectEntriesPage() throws Exception {
 			_siteScopedObjectEntry1.getObjectEntryId());
 	}
 
+	@Test
+	public void testGetScopeScopeKeyPageWithSiteExternalReferenceCode()
+		throws Exception {
+
+		Group group = _addObjectEntryAndGetGroup();
+
+		JSONObject jsonObject = _invoke(
+			Http.Method.GET, group.getExternalReferenceCode(), null);
+
+		JSONArray itemsJSONArray = jsonObject.getJSONArray("items");
+
+		Assert.assertEquals(1, itemsJSONArray.length());
+
+		JSONObject itemJSONObject = itemsJSONArray.getJSONObject(0);
+
+		Assert.assertEquals(
+			itemJSONObject.getLong("id"),
+			_siteScopedObjectEntry1.getObjectEntryId());
```

Comment:
I see that this code block is testing the method `getScopeScopeKeyPage`. Why not extract this into a separate test method that focuses solely on this logic, and include the three scenarios (ERC, ID, and name)?